context/app/static/js/components/workspaces/WorkspaceField/WorkspaceField.tsx

```diff
@@ -24,6 +24,12 @@ function WorkspaceField<FormType extends FieldValues>({
     defaultValue: value,
   });
 
+  useEffect(() => {
+    if (value) {
+      field.onChange(value);
+    }
+  }, [value, field]);
+
```

Comment:
I'm a bit curious what introduced this bug. We were only experiencing it with the protected datasets field, but not the other instances of `WorkspaceField`.

Reply:
From what I could tell, the other instances of WorkspaceField are only related to the workspace name field, which defaults to an empty string - this one defaults to the protected dataset ids, which are initially blank. This potentially changes later on depending on which datasets are selected, but the form wasn't updating the default field value with the new IDs after the initial render.

Reply:
Well it definitely looks like I introduced this bug (sorry)! Can you try providing an optional initial value for `protected-datasets` to `useCreateWorkspaceForm` like we do in `/workspaces/AddDatasetsFromSearchDialog/hooks.ts`?

Reply:
Yes, that works - however, I'm still seeing a need for a `useEffect` function in `hooks.ts` to monitor when the initial value changes from an empty string to a list of IDs. I'll push my changes in one moment.